### PR TITLE
Implementation for Issue #422

### PR DESCRIPTION
**Pull request by TricksterGuy**
Let me know if there are any problems. I just used WeighDuplicatesTogether as a framework to implement this.

### CODE REVIEW

**Comment by TricksterGuy** [comment]
The changes are made as requested.

**Review by tom-overton — CHANGES_REQUESTED** [review]
Thanks, just one last thing:

File: src/com/dabomstew/pkrandom/Settings.java [hunk 1]
```diff
@@ -869,6 +872,7 @@ public static Settings fromString(String settingsString) throws UnsupportedEncod
         settings.setBanIrregularAltFormes(restoreState(data[49], 3));
 
         settings.setEliteFourUniquePokemonNumber(data[50] & 0x7);
+        settings.setEnsureTwoAbilities(restoreState(data[51], 7));
```
[the remark this answers is not in the document]
Comment (TricksterGuy) [reply src/com/dabomstew/pkrandom/Settings.java]: snap forgot to remove that fixed!

**Review by tom-overton — APPROVED** [review]

**Review by Ajarmar — APPROVED** [review]
Looks good, I'll merge it. Thanks for the contribution!

**Pull request merged**